kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/SessionSerializer.java

```diff
@@ -442,18 +451,20 @@ private String getClusterKey(Map<String, Object> attributes) {
     }
 
     private Map<String, Object> doDeserialize(SessionInfo sessionInfo,
-            String sessionId) throws IOException, ClassNotFoundException {
+            String sessionId) throws Exception {
         byte[] data = sessionInfo.getData();
         long start = System.currentTimeMillis();
 
         // Is this needed?
         ClassLoader contextLoader = Thread.currentThread()
                 .getContextClassLoader();
         ByteArrayInputStream in = new ByteArrayInputStream(data);
-        Map<String, Object> attributes;
+        Map<String, Object> attributes = new HashMap<>();
         try (TransientInjectableObjectInputStream inStream = new TransientInjectableObjectInputStream(
                 in, handler)) {
             attributes = inStream.readWithTransients();
+        } catch (Exception ex) {
+            sessionSerializationCallback.onDeserializationError(ex);
```

Comment:
Same here, if we continue the execution because the user callback does not re-throw we will provide a "wrong" session, that may cause bugs difficult to detect.

Reply:
Added the exception to be thrown.